### PR TITLE
chore: Upgrade to Typescript `~2.6.1`

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

File: src/agent/debuglet.ts [hunk 1]
```diff
@@ -420,20 +421,20 @@ export class Debuglet extends EventEmitter {
     }
   }
 
-  static getProjectIdFromMetadata(): Promise<string> {
-    return new Promise<string>((resolve, reject) => {
+  static getProjectIdFromMetadata(): Promise<string|undefined> {
```
Comment (ofrobots) [inline src/agent/debuglet.ts]: nit: drop inferable return type.

File: src/agent/debuglet.ts [hunk 2]
```diff
@@ -420,20 +421,20 @@ export class Debuglet extends EventEmitter {
...
-          'project-id', (err: Error, res: any, projectId: string) => {
+          'project-id', (err: Error|null, res?: request.RequestResponse, projectId?: string) => {
             err ? reject(err) : resolve(projectId);
           });
     });
   }
 
-  static getClusterNameFromMetadata(): Promise<string> {
-    return new Promise<string>((resolve, reject) => {
+  static getClusterNameFromMetadata(): Promise<string|undefined> {
```
Comment (ofrobots) [inline src/agent/debuglet.ts]: nit: drop inferable return type.

File: src/agent/debuglet.ts [hunk 3]
```diff
@@ -535,7 +536,7 @@ export class Debuglet extends EventEmitter {
       that.logger.info('Fetching breakpoints');
       // TODO: Address the case when `that.debuggee` is `null`.
       that.controller.listBreakpoints(
-          (that.debuggee as Debuggee), function(err: Error, response, body) {
+          (that.debuggee as Debuggee), function(err: Error|null, response, body) {
```
Comment (ofrobots) [inline src/agent/debuglet.ts]: question: could the err type annotation be omitted

File: src/agent/debuglet.ts [hunk 4]
```diff
@@ -420,20 +421,20 @@ export class Debuglet extends EventEmitter {
...
   }
 
-  static getClusterNameFromMetadata(): Promise<string> {
-    return new Promise<string>((resolve, reject) => {
+  static getClusterNameFromMetadata(): Promise<string|undefined> {
+    return new Promise<string|undefined>((resolve, reject) => {
       metadata.instance(
           'attributes/cluster-name',
-          (err: Error, res: any, clusterName: string) => {
+          (err: Error|null, res?: request.RequestResponse, clusterName?: string) => {
```
Comment (ofrobots) [inline src/agent/debuglet.ts]: question: would if work if you dropped the type annotations for parameters?

File: src/agent/debuglet.ts (hunk 1 above)
Comment (DominicKramer) [reply src/agent/debuglet.ts]: Done

File: src/agent/debuglet.ts (hunk 2 above)
Comment (DominicKramer) [reply src/agent/debuglet.ts]: Done

File: src/agent/debuglet.ts (hunk 4 above)
Comment (DominicKramer) [reply src/agent/debuglet.ts]: Yes. Removed.

File: src/agent/debuglet.ts (hunk 3 above)
Comment (DominicKramer) [reply src/agent/debuglet.ts]: Yes. Removed.

File: test/test-controller.ts [hunk 5]
```diff
@@ -79,11 +80,12 @@ describe('Controller API', function() {
          });
          const controller = new Controller(fakeDebug);
          controller.register(
-             debuggee, function(err: Error, result: {debuggee: Debuggee}) {
+             debuggee, function(err: Error|null, result?: {debuggee: Debuggee}) {
```
Comment (ofrobots) [inline test/test-controller.ts]: nit: and here

File: test/test-controller.ts [hunk 6]
```diff
@@ -57,9 +57,10 @@ describe('Controller API', function() {
       const controller = new Controller(fakeDebug);
       // TODO: Determine if this type signature is correct.
       controller.register(
-          debuggee, function(err: Error|null, result: {debuggee: Debuggee}) {
+          debuggee, function(err: Error|null, result?: {debuggee: Debuggee}) {
```
Comment (ofrobots) [inline test/test-controller.ts]: nit: would things work if you dropped the parameter type definitions?

File: test/test-controller.ts [hunk 7]
```diff
@@ -232,11 +234,12 @@ describe('Controller API', function() {
         controller.listBreakpoints(
             debuggee,
             function(
-                err: Error|null, response: http.ServerResponse,
-                result: stackdriver.ListBreakpointsResponse) {
+                err: Error|null, response?: http.ServerResponse,
```
Comment (ofrobots) [inline test/test-controller.ts]: nit: and here.

File: test/test-max-data-size.ts [hunk 8]
```diff
@@ -75,10 +75,10 @@ describe('maxDataSize', function() {
       location: breakpointInFoo.location
     } as stackdriver.Breakpoint;
     // TODO: Determine how to remove this cast to any.
-    api.set(bp, function(err1: Error) {
+    api.set(bp, function(err1: Error|null) {
```
Comment (ofrobots) [inline test/test-max-data-size.ts]: and here

File: test/test-max-data-size.ts [hunk 9]
```diff
@@ -109,9 +109,9 @@ describe('maxDataSize', function() {
       id: breakpointInFoo.id,
       location: breakpointInFoo.location
     } as stackdriver.Breakpoint;
-    api.set(bp, function(err1: Error) {
+    api.set(bp, function(err1: Error|null) {
```
Comment (ofrobots) [inline test/test-max-data-size.ts]: and here.

File: test/test-controller.ts (hunk 6 above)
Comment (DominicKramer) [reply test/test-controller.ts]: Done

File: test/test-controller.ts (hunk 5 above)
Comment (DominicKramer) [reply test/test-controller.ts]: Done

File: test/test-controller.ts (hunk 7 above)
Comment (DominicKramer) [reply test/test-controller.ts]: Done

File: test/test-max-data-size.ts (hunk 8 above)
Comment (DominicKramer) [reply test/test-max-data-size.ts]: Done

File: test/test-max-data-size.ts (hunk 9 above)
Comment (DominicKramer) [reply test/test-max-data-size.ts]: Done

**Pull request merged**